### PR TITLE
Refactor Failures

### PR DESCRIPTION
**Pull request by AydinHassan**

### CODE REVIEW

**Comment by AydinHassan** [comment]
Need to fix result renderer

**Comment by AydinHassan** [comment]
@mikeymike Any chance of a quick review? It's mostly just cleaning up some things, I need it to add a multiple request feature to the `CgiCheck`

File: src/Check/PhpLintCheck.php [hunk 1]
```diff
@@ -27,10 +35,10 @@ public function check(ExerciseInterface $exercise, $fileName)
         $process->run();
 
         if ($process->isSuccessful()) {
-            return new Success('PHP Code Check');
+            return new Success($this);
         }
 
-        return new Failure('PHP Code Check', $process->getErrorOutput());
+        return new Failure($this, $process->getErrorOutput());
```
Comment (mikeymike) [inline src/Check/PhpLintCheck.php]: should this not be Failure::withReason ?

**Comment by mikeymike** [comment]
Looks good dude, much nicer with the interfaces :+1:

File: src/Check/PhpLintCheck.php (hunk 1 above)
Comment (AydinHassan) [inline src/Check/PhpLintCheck.php]: Yeah good catch. I'll amend it in a later PR, it can work like that also.

**Pull request merged**